### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by mads256h**
Seperate out IRoomRepository and ITrackerRepository

We are no longer using the database objects for serialization and deserialization on the REST Api.
Now we have specific objects that represent the requests. This means we no longer have data inside that are not used and we no longer have to juggle entity framework and json serialization.

We now also have response objects.
These give the user of the api more information about the status of the request.
For example adding a room now gives the user the id of the newly added room.
Another example is that adding a tracker to a room is now fallible. If the room id the user tries to add a tracker to does not exist we return that the request was unsuccessful.

### CODE REVIEW

**Review by Taoshix — APPROVED** [review]
Much better now

**Review by Taoshix — APPROVED** [review]
Everything is json constructor now, had to refresh to see it

File: CentralHub.Api.Tests/TrackersControllerTests.cs [hunk 1]
```diff
@@ -1,136 +1,171 @@
...
+using CentralHub.Api.Model.Requests;
 using CentralHub.Api.Services;
 
 namespace CentralHub.Api.Tests;
 
 public class TrackersControllerTests
 {
-    private RoomRepository _roomRepository;
-    private ILocalizationTargetService _localizationTargetService;
+    private ILocalizationService _localizationService;
```
Comment (Casper-NS) [inline CentralHub.Api.Tests/TrackersControllerTests.cs]: this is never used, plz remove

File: CentralHub.Api/Controllers/LocalizationController.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    public LocalizationController(ILocalizationService localizationService)
+    {
+        _localizationService = localizationService;
+    }
+
+    [HttpPost("measurements/add")]
+    public async Task AddMeasurements(AddMeasurementsRequest addMeasurementsRequest, CancellationToken token)
+    {
+        _localizationService.AddMeasurements(
```
Comment (Casper-NS) [inline CentralHub.Api/Controllers/LocalizationController.cs:21]: await?

File: CentralHub.Api/Controllers/TrackerController.cs [hunk 3]
```diff
@@ -9,51 +12,76 @@ namespace CentralHub.Api.Controllers;
 public class TrackerController : ControllerBase
 {
     private readonly IRoomRepository _roomRepository;
-    private readonly ILocalizationTargetService _localizationTargetService;
+    private readonly ILocalizationService _localizationService;
```
Comment (Casper-NS) [inline CentralHub.Api/Controllers/TrackerController.cs:15]: Not used.

**Review by Casper-NS — CHANGES_REQUESTED** [review]
Minor changes needed, but otherwise lgtm

File: CentralHub.Api/Controllers/LocalizationController.cs (hunk 2 above)
Comment (mads256h) [reply CentralHub.Api/Controllers/LocalizationController.cs:21]: Det kan man ikke

File: CentralHub.Api/Controllers/TrackerController.cs (hunk 3 above)
Comment (mads256h) [reply CentralHub.Api/Controllers/TrackerController.cs]: Fixed

**Review by Casper-NS — APPROVED** [review]
Lgtm

**Pull request merged**